tests_integration/test_integration.py

```diff
@@ -0,0 +1,298 @@
+        args = ["./mock_proxy_start.sh"]
+        timeout = 60
+        popen_kwargs = {"cwd": TESTS_INTEGRATION_DIR}
+        env = os.environ | {
+            "SCENARIO_ID": str(scenario_id),
+            "PYTHONPATH": str(CF_SCRIPTS_ROOT_DIR.resolve()),
+        }
+
+        def startup_check(self):
+            return is_proxy_running(port=8080)
+
+    xprocess.ensure("mitmproxy", MitmproxyStarter)
+
+    yield
+
+    xprocess.getinfo("mitmproxy").terminate()
+
+
+@contextlib.contextmanager
+def in_fresh_cf_graph():
+    """
+    Context manager to execute code within the context with a new clone
+    of cf-graph in a temporary directory.
+    """
+    old_working_dir = os.getcwd()
+
+    with tempfile.TemporaryDirectory() as tmpdir_s:
+        tmpdir = Path(tmpdir_s)
+
+        cf_graph_repo = settings().graph_github_backend_repo
+
+        # --depth=5 is the same value as used in prod (see autotick-bot/install_bot_code.sh)
+        subprocess.run(
+            [
+                "git",
+                "clone",
+                "--depth=5",
+                f"https://github.com/{cf_graph_repo}.git",
+                "cf-graph",
+            ],
+            check=True,
+            cwd=tmpdir,
+        )
+
+        cf_graph_dir = tmpdir / "cf-graph"
+
+        subprocess.run(
+            [
+                "git",
+                "config",
+                "user.name",
+                "regro-cf-autotick-bot-staging",
+            ],
+            check=True,
+            cwd=cf_graph_dir,
+        )
+
+        subprocess.run(
+            [
+                "git",
+                "config",
+                "user.email",
+                "regro-cf-autotick-bot-staging@users.noreply.github.com",
+            ],
+            check=True,
+            cwd=cf_graph_dir,
+        )
+
+        subprocess.run(
+            [
+                "git",
+                "config",
+                "pull.rebase",
+                "false",
+            ],
+            check=True,
+            cwd=cf_graph_dir,
+        )
+
+        subprocess.run(
+            ["git", "config", 'http."https://github.com".proxy', '""'], check=True
+        )
+
+        os.chdir(cf_graph_dir)
+        yield
+
+    os.chdir(old_working_dir)
+
+
+@contextlib.contextmanager
+def mitmproxy_env():
+    """
+    Set environment variables for bot steps that should be piped through mitmproxy.
+    """
+    old_env = os.environ.copy()
+
+    os.environ["http_proxy"] = "http://127.0.0.1:8080"
+    os.environ["https_proxy"] = "http://127.0.0.1:8080"
+    os.environ["CF_FEEDSTOCK_OPS_CONTAINER_PROXY_MODE"] = "true"
+
+    yield
+
+    os.environ.clear()
+    os.environ.update(old_env)
+
+
+def invoke_bot_command(args: list[str]):
+    """
+    Invoke the bot command with the given arguments.
+    """
+    from conda_forge_tick import cli
+
+    cli.main(args, standalone_mode=False)
+
+
+@pytest.mark.parametrize("use_containers", [False, True])
+def test_scenario(
```

Comment:
This deserves a docstring with a brief overview and pointers to where to find more details

Reply:
done in 5dd850d01f8e334af3b413e8981f49c9257cfdc2